front/app/containers/IdeasEditPage/WithJSONForm/index.tsx

```diff
@@ -0,0 +1,167 @@
+import useAuthUser from 'hooks/useAuthUser';
+import useProject from 'hooks/useProject';
+import usePhases from 'hooks/usePhases';
+import useInputSchema from 'hooks/useInputSchema';
+import { getInputTerm } from 'services/participationContexts';
+
+import { FormattedMessage } from 'utils/cl-intl';
+import messages from '../messages';
+import ideaFormMessages from 'containers/IdeasNewPage/messages';
+
+import Form, { AjvErrorGetter, ApiErrorGetter } from 'components/Form';
+
+import PageContainer from 'components/UI/PageContainer';
+import FullPageSpinner from 'components/UI/FullPageSpinner';
+import { updateIdea } from 'services/ideas';
+import { geocode } from 'utils/locationTools';
+import useIdea from 'hooks/useIdea';
+import IdeasEditMeta from '../IdeasEditMeta';
+import { usePermission } from 'services/permissions';
+import { getFieldNameFromPath } from 'utils/JSONFormUtils';
+
+const IdeasEditPageWithJSONForm = ({ params: { ideaId } }: WithRouterProps) => {
+  const previousPathName = useContext(PreviousPathnameContext);
+  const authUser = useAuthUser();
+  const idea = useIdea({ ideaId });
+  const project = useProject({
+    projectId: isNilOrError(idea) ? null : idea.relationships.project.data.id,
+  });
+
+  const phases = usePhases(project?.id);
+  const { schema, uiSchema, inputSchemaError } = useInputSchema(project?.id);
+  const permisison = usePermission({
+    item: isNilOrError(idea) ? null : idea,
+    action: 'edit',
+    context: idea,
+  });
+
+  useEffect(() => {
+    if (!isNilOrError(idea) && !permisison) {
+      clHistory.replace(previousPathName || (!authUser ? '/sign-up' : '/'));
+    }
+  }, [authUser, idea, previousPathName, permisison]);
+
+  const initialFormData =
+    isNilOrError(idea) || !schema
+      ? null
+      : Object.fromEntries(
+          Object.keys(schema.properties).map((prop) => {
+            if (idea.attributes?.[prop]) {
+              return [prop, idea.attributes?.[prop]];
+            } else if (
+              prop.endsWith('_ids') &&
+              Array.isArray(
+                idea.relationships?.[prop.replace('_ids', 's')]?.data
+              )
+            ) {
+              return [
+                prop,
+                idea.relationships?.[prop.replace('_ids', 's')].data.map(
+                  (rel) => rel.id
+                ),
+              ];
+            } else return [prop, undefined];
+          })
+        );
+
+  const onSubmit = async (data) => {
+    let location_point_geojson;
+
+    if (data.location_description && !data.location_point_geojson) {
+      location_point_geojson = await geocode(data.location_description);
+    }
+
+    const idea = await updateIdea(ideaId, {
+      ...data,
+      location_point_geojson,
+      project_id: project?.id,
+      publication_status: 'published',
+    });
+
+    clHistory.push({
+      pathname: `/ideas/${idea.data.attributes.slug}`,
+    });
+  };
+
+  const getApiErrorMessage: ApiErrorGetter = useCallback(
```

Comment:
The form taks two translation function for errors, one for api and one for the front-end errors called ajvErrors